internal/command/instance_features.go

```diff
@@ -0,0 +1,67 @@
+package command
+
+import (
+	"context"
+
+	"github.com/zitadel/zitadel/internal/api/authz"
+	"github.com/zitadel/zitadel/internal/command/preparation"
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/eventstore"
+	"github.com/zitadel/zitadel/internal/feature"
+	"github.com/zitadel/zitadel/internal/repository/feature/feature_v2"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+type InstanceFeatures struct {
+	LoginDefaultOrg                 *bool
+	TriggerIntrospectionProjections *bool
+	LegacyIntrospection             *bool
+}
+
+func appendNonNilFeature[T any](ctx context.Context, cmds []eventstore.Command, aggregate *feature_v2.Aggregate, value *T, eventType eventstore.EventType) []eventstore.Command {
+	if value != nil {
+		cmds = append(cmds, feature_v2.NewSetEvent[T](ctx, aggregate, eventType, *value))
+	}
+	return cmds
+}
+
+func (c *Commands) SetInstanceFeatures(ctx context.Context, f *InstanceFeatures) (*domain.ObjectDetails, error) {
+	instanceID := authz.GetInstance(ctx).InstanceID()
+	cmds := instanceFeatureSetCommands(ctx, instanceID, f)
+	if len(cmds) == 0 {
+		return nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-Gie6U", "Errors.NoChangesFound")
+	}
```

Comment:
shouldn't there be a writemodel getting the current state (in this case instance features) and check if / which events need to be created
in case none needs to be created, i'd simply return the current state as details like here https://github.com/zitadel/zitadel/blob/f680dd934df9c1ff2b3b91b4565997cb2c2db6a2/internal/command/instance_idp.go#L45-L52

Reply:
From a business logic point of view the current state is not important, hence I went for a direct-push approach. But I agree with your comment as I see that the response will remain empty and it may produce unneeded events that the system needs to handle. I will create a write model.